libcalico-go/lib/validator/v3/validator.go

```diff
@@ -1567,6 +1567,18 @@ func validateTier(structLevel validator.StructLevel) {
 		}
 	}
 
+	if tier.Name == names.AdminNetworkPolicyTierName {
```

Comment:
We don't validate this on read. In the Tier implementation of Clientv3, we only call validator on create and update.
https://github.com/projectcalico/calico/blob/1e410644fc11608d98cc3810c63a4c4d7c8f6f95/libcalico-go/lib/clientv3/tier.go#L146

Reply:
Felix/Typha both validate in their calc graphs, I think.

Reply:
Right. It seems in Typha we also validate in reads: https://github.com/projectcalico/calico/blob/4ad72b7c787a714403febb2ad72c5947b94d3647/typha/pkg/calc/validation_filter.go#L64
Also in Felix: https://github.com/projectcalico/calico/blob/4ad72b7c787a714403febb2ad72c5947b94d3647/felix/calc/validation_filter.go#L67

Reply:
Originally the check was like this: https://github.com/tigera/calico-private/blame/d7f58624426ef41c64b6981b6017cf2956b50a25/libcalico-go/lib/clientv3/tier.go#L51

However, we changed it to v3 validator since the return error code, `Operation not supported`, is not technically correct, as the create/update operations are supported. It just feels the right place for these type of checks is in validator.

Reply:
Will this be an issue if create and update does not allow the incorrect value for AdminNeworkPolicy tier in first place?

Reply:
We can also use `DefaultTierFields` here: https://github.com/projectcalico/calico/blob/master/libcalico-go/lib/resources/tier.go#L21
to make sure values are correct. This function is called by k8s client: https://github.com/projectcalico/calico/blob/master/libcalico-go/lib/backend/k8s/resources/tiers.go

Reply:
btw, we create adminneworkpolicy tier ourselves at the beginning of the day here: https://github.com/projectcalico/calico/blob/018190a67bef887320db854094b0866a95f35284/libcalico-go/lib/clientv3/client.go#L421